### DISCUSSION TITLE
exhaustive switch support

### DISCUSSION THREAD

**Discussion by ellemedit** [TanStack/query#923] [General] [answered]
```
type T =  "a" | "b" | "c";

const f = (t: T) => {
    switch (t) {
        case 'a': {
            return 0;
        }
        case 'b': {
            return 2;
        }
        case 'c': {
            return 1;   
        }
    }
}

enum T2 {
    a = "a",
    b = "b",
    c = "c"
}

const f2 = (t: T2) => {
    switch (t) {
        case 'a': {
            return 0;
        }
        case 'b': {
            return 2;
        }
        case 'c': {
            return 1;   
        }
    }
    return null;
}
```

f1 return type inferred as `0 | 1 | 2`, but f2 inferred as `0 | 1 | 2 | null`. TypeScript is not smart enough to infer enum type for exhaustive switch. Previous type interface supported exhaustive switch, thus developers can safely use useQuery via status. Could we rollback this?

**Comment by tannerlinsley** [comment]
Short answer is... probably not any time soon. @boschni can elaborate more on this.

**Comment by boschni** [comment, marked as answer]
Hi @Beingbook! If you specify the actual enum in the switch cases then the exhaustive check should work :)

**Reply by ellemedit** [reply]
yes, it works with enum. but I think it might be verbose.
